modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/rel/HashAggregateNode.java

```diff
@@ -51,10 +51,14 @@ public class HashAggregateNode<RowT> extends AbstractNode<RowT> implements Singl
 
     private final RowFactory<RowT> rowFactory;
 
-    private final ImmutableBitSet grpSet;
+    /** A bit set that contains fields included in all grouping sets. */
+    private final ImmutableBitSet allFields;
 
     private final List<Grouping> groupings;
 
+    /** Mapping between group columns and their positions inside an input row.*/
+    private final Mapping mapping;
```

Comment:
Can't find a single test in which this mapping is not an identity mapping (0->0, 1->1, etc)... Do you have such test in mind? 

The reason I want to see it in action is that I have feeling you are trying to fix broken plan in runtime with this mapping. We definitely don't want to do this.

Reply:
@korlov42 

We need this mapping because the positions between the same columns on MAP phase and REDUCE phase are different. Example.

> sql("CREATE TABLE testMe (a INTEGER, b INTEGER, s VARCHAR);");
> sql("INSERT INTO testMe VALUES (11, 1, 'hello'), (12, 2, 'world'), (11, 3, NULL)");
> sql("INSERT INTO testMe VALUES (11, 3, 'hello'), (12, 2, 'world'), (10, 5, 'ahello'), (13, 6, 'world')");
            
Query:

>SELECT COUNT(a), COUNT(DISTINCT(b)) FROM testMe

Logical plan:
> LogicalAggregate(group=[{}], EXPR$0=[COUNT($0)], EXPR$1=[COUNT(DISTINCT $1)]), id = 5
>  LogicalProject(A=[$0], B=[$1]), id = 4
>    IgniteLogicalTableScan(table=[[PUBLIC, TESTME]]), id = 1

Optimized plan:

>IgniteProject(EXPR$0=[CAST($0):BIGINT NOT NULL], EXPR$1=[$1]), id = 128
>  IgniteProject(EXPR$0=[$0], EXPR$1=[CAST($1):BIGINT NOT NULL]), id = 127
>    IgniteReduceSortAggregate(group=[{}], EXPR$0=[$SUM0($0)], agg#1=[$SUM0($1)], collation=[[]]), id = 126
>      IgniteMapSortAggregate(group=[{}], EXPR$0=[$SUM0($1)], EXPR$1=[COUNT($0)], collation=[[]]), id = 125
>        IgniteProject(B=[$0], EXPR$0=[CAST($1):BIGINT NOT NULL]), id = 124
>          IgniteReduceHashAggregate(group=[{1}], agg#0=[$SUM0($1)]), id = 123
>            IgniteMapHashAggregate(group=[{1}], EXPR$0=[COUNT($0)]), id = 122
>              IgniteTableScan(table=[[PUBLIC, TESTME]], requiredColumns=[{0, 1}]), id = 59


If we look at the first Aggregate (the aggregate that takes a table as it's input)
>        IgniteReduceHashAggregate(group=[{1}], agg#0=[$SUM0($1)]), id = 123
>            IgniteMapHashAggregate(group=[{1}], EXPR$0=[COUNT($0)]), id = 122

Aggregate:

> input: RecordType(INTEGER A, INTEGER B)
> grouping sets: [{1}]
> output:  RecordType(INTEGER B, BIGINT _ACC0, TINYINT _GROUP_ID)

MAP 
>input: RecordType(INTEGER A, INTEGER B)
>output: RecordType(INTEGER B, BIGINT _ACC0, TINYINT _GROUP_ID)
>grouping sets: [{1}]

REDUCE 
>input:  RecordType(INTEGER B, BIGINT _ACC0, TINYINT _GROUP_ID)
>output: RecordType(INTEGER f0, BIGINT f1)
>grouping sets: [{1}]

We see that Column A is assigned index to 0, column B is assigned to 1. Grouping sets are `{{1}} ` where `1` is a position in MAP's input. 
The ticket states that MAP/REDUCE should use the same logic (because they are almost identical). To achieve that we need a mapping between for columns used on REDUCE phase and for MAP phase we use identity mapping.
Output of this aggregate does not produce `A` but REDUCE phase uses `B` on the second phase in both it's input and its output. So we map A to 0, 1 to B on MAP phase, and B to 0 on map phase.

Reply:
Updated grouping set field index computation on REDUCE phase and removed `mapping` property of `HashAggregateNode`.